app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -240,16 +240,41 @@ impl List {
             }
         }
         self.local_scope.update_sorting(components_order);
-        let top_modules_iter = self.module_groups.values().filter(|g| g.is_top_module);
-        let mut top_mdl_bld = component::group::AlphabeticalListBuilder::default();
-        top_mdl_bld.extend(top_modules_iter.clone().map(|g| g.content.clone_ref()));
-        let mut top_mdl_flat_bld = component::group::AlphabeticalListBuilder::default();
-        top_mdl_flat_bld.extend(top_modules_iter.filter_map(|g| g.flattened_content.clone()));
         let favorites = self.build_favorites_and_add_to_all_components();
+
+        // Separate top module groups by namespace.
+        let mut top_modules_by_namespace: BTreeMap<_, Vec<&ModuleGroups>> = BTreeMap::new();
+        for group in self.module_groups.values().filter(|g| g.is_top_module) {
+            let namespace = group.qualified_name.project().namespace.clone();
+            top_modules_by_namespace.entry(namespace).or_insert_with(default).push(group);
+        }
+        // Create alphabetical lists of top modules per section.
+        let mut top_modules = Vec::new();
+        let mut top_modules_flattened = Vec::new();
+        let mut top_section_names = Vec::new();
+        for (namespace, modules) in top_modules_by_namespace {
+            let top_modules_iter = modules.iter();
+            let mut top_mdl_bld = component::group::AlphabeticalListBuilder::default();
+            top_mdl_bld.extend(top_modules_iter.clone().map(|g| g.content.clone_ref()));
+            top_modules.push(top_mdl_bld.build());
+            let mut top_mdl_flat_bld = component::group::AlphabeticalListBuilder::default();
+            top_mdl_flat_bld.extend(top_modules_iter.filter_map(|g| g.flattened_content.clone()));
+            top_modules_flattened.push(top_mdl_flat_bld.build());
+            top_section_names.push(namespace);
+        }
+        let top_section_positions = Rc::new(
+            top_section_names.iter().enumerate().map(|(pos, name)| (name.clone(), pos)).collect(),
+        );
+        let top_modules = Rc::new(top_modules);
+        let top_modules_flattened = Rc::new(top_modules_flattened);
+        let top_section_names = Rc::new(top_section_names);
```

Comment:
Having a separate "Section" structure, we could also have "builder::Section" with the logic inside for loop. I think the code will be a bit cleaner.